src/main/java/com/internship/backend/service/TennisCourtService.java

```diff
@@ -20,14 +20,20 @@ public class TennisCourtService {
     @Autowired
     private LocationRepository locationRepository;
 
+    @Autowired
+    private IdGeneratorService idGeneratorService;
+
+
     public List<TennisCourt> getAllTennisCourts(){
         return tennisCourtRepository.findAll();
     }
 
     public TennisCourt addTennisCourt(TennisCourt tennisCourt) throws TennisCourtAlreadyExistsException {
+        tennisCourt.setId(idGeneratorService.getCurrentId());
```

Comment:
I think you can use another much simpler method to generate tennis courts ids and exclude the possibility of having several courts with the same id. I think you can use @Id and @GeneratedValue annotations and in this way you will be able to simplify the logic of adding a court.